src/Workspaces/Core/Portable/Utilities/ParameterName.cs

```diff
@@ -51,7 +52,7 @@ public ParameterName(string nameBasedOnArgument, bool isFixed, NamingRule parame
             {
                 // Otherwise, massage it a bit to be a more suitable match for
                 // how people actually writing parameters.
-                var trimmed = nameBasedOnArgument.TrimStart('_');
+                var trimmed = NamingStyle.StripCommonPrefixes(nameBasedOnArgument, out var _);
```

Comment:
1. consider `out _`.
2. consider having an overload without the second parameter at all.